src/queue.jl

```diff
@@ -9,8 +9,18 @@ end
 
 Create a `Queue` object containing elements of type `T`.
 """
-Queue(ty::Type{T}) where {T} = Queue(Deque{T}())
-Queue(ty::Type{T}, blksize::Integer) where {T} = Queue(Deque{T}(blksize))
+Queue{T}() where {T} = Queue(Deque{T}())
+Queue{T}(blksize::Integer) where {T} = Queue(Deque{T}(blksize))
+
+function Queue(ty::Type{T}) where {T}
+    Base.depwarn("Queue(T) is deprecated, use Queue{T}() instead")
```

Comment:
I think you can just use
```julia
@deprecate Queue(::Type{T}) where {T} Queue{T}()
```
If that doesn't work, `depwarn` requires a second argument, so this should instead be
```julia
    Base.depwarn("Queue(T) is deprecated, use Queue{T}() instead", :Queue)
```